src/main/java/org/minions/devfund/chelo/WhackAMole.java

```diff
@@ -0,0 +1,259 @@
+            if (place(x, y)) {
+                this.moleGrid[x][y] = 'M';
+                molesLeft--;
+            }
+        }
+    }
+
+    /**
+     * Method given a location, returns if availability.
+     * @param x coordinate
+     * @param y coordinate
+     * @return status of the coordinate
+     */
+    boolean place(int x, int y) {
+        if (this.moleGrid[x][y] != 'M') {
+            this.moleGrid[x][y] = 'M';
+            this.molesLeft++;
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * A method given a location, take a whack at that location.
+     * If that location contains a mole, the score, number of attemptsLeft, and molesLeft is updated.
+     * If that location does not contain a mole, only attemptsLeft is updated.
+     * @param x coordinate
+     * @param y coordinate
+     */
+    void whack(int x, int y) {
+        if (moleGrid[x][y] == 'M') {
+            moleGrid[x][y] = 'W';
+            score++;
+            molesLeft--;
+            attemptsLeft--;
+        } else {
+            attemptsLeft--;
+        }
+    }
+
+    /**
+     * A method to print the grid without showing where the moles are.
+     * For every spot that has recorded a whacked mole, print out a W, or * otherwise.
+     */
+    void printGridToUser() {
+        for (int i = 0; i < moleGrid.length; i++) {
+            for (int j = 0; j < moleGrid.length; j++) {
+                if (moleGrid[i][j] == 'M') {
+                    System.out.print("[*]");
+                } else {
+                    System.out.print("[" + moleGrid[i][j] + "]");
+                }
+            }
+            System.out.println(" ");
+        }
+    }
+
+    /**
+     * A method to print the grid without showing where the moles are.
+     * For every spot that has recorded a whacked mole print out a W, or * otherwise.
+     */
+    void printGrid() {
+        for (int i = 0; i < moleGrid.length; i++) {
+            for (int j = 0; j < moleGrid.length; j++) {
+                System.out.print("[" + moleGrid[i][j] + "]");
+            }
+            System.out.println(" ");
+        }
+    }
+
+    /**\
+     * A method to return Attempts Left value.
+     * @return remaining attempts
+     */
+    int getAttemptsLeft() {
+        return attemptsLeft;
+    }
+
+    /**
+     * A method to return moles left value.
+     * @return remaining moles
+     */
+    int getMolesLeft() {
+        return  molesLeft;
+    }
+
+    /**
+     * A method to print the current status for.
+     * Score,
+     * Attempts Left,
+     * Moles Left.
+     */
+    public void printStatus() {
+        System.out.println("Score: " + this.score
+                + ", Attempts left: " + this.attemptsLeft
+                + ", Moles left: " + this.molesLeft);
+    }
+
+    /**
+     * Main method to run the program.
+     * @param args main method
+     */
+    public static void main(final String[] args) {
```

Comment:
You may delete this static main since it is not necessary in edx

Reply:
I did not upload MAIN method in EDX, but here it's required to pass 'gradle clean check'